python/ray/train/_internal/schema.py

```diff
@@ -0,0 +1,63 @@
+from typing import List, Optional
+
+from ray._private.pydantic_compat import BaseModel, Field
+from ray.util.annotations import DeveloperAPI
+
+try:
+    import pydantic  # noqa: F401
+except ImportError:
+    raise ModuleNotFoundError(
+        "pydantic isn't installed. "
+        "To install pydantic, please run 'pip install pydantic'"
+    )
+
+
+@DeveloperAPI
+class TrainWorkerInfo(BaseModel):
+    """Metadata of a Ray Train worker."""
+
+    actor_id: str = Field(description="Actor ID of the worker.")
+    world_rank: int = Field(description="World rank.")
+    local_rank: int = Field(description="Local rank.")
+    node_rank: int = Field(description="Node rank.")
+    gpu_ids: Optional[List[str]] = Field(
+        description="A list of GPU ids allocated to that worker."
+    )
+    node_id: Optional[str] = Field(
+        description="ID of the node that the worker is running on."
+    )
+    node_ip: Optional[str] = Field(
+        description="IP address of the node that the worker is running on."
+    )
+    pid: Optional[str] = Field(description="PID of the worker.")
+
+
+@DeveloperAPI
+class TrainDatasetInfo(BaseModel):
```

Comment:
Make dataset a pydantic class, so that we can add per-worker shards info here in the future.